graphql_compiler/compiler/ir_lowering_match/ir_lowering.py

```diff
@@ -93,95 +93,47 @@ def visitor_fn(expression):
     return new_ir_blocks
 
 
-def lower_string_operators(ir_blocks):
-    """Lower Filter blocks that use the "has_substring", "starts_with", or
-    "ends_with" operation into MATCH-representable form."""
+def prepend_wildcard(expression):
+    return BinaryComposition(
+        u'+',
+        Literal('%'),
+        expression
+    )
+
+
+def append_wildcard(expression):
+    return BinaryComposition(
+        u'+',
+        expression,
+        Literal('%')
+    )
 
+
+def lower_string_operators(ir_blocks):
+    """Lower Filters with "has_substring", "starts_with", or "ends_with" operation into MATCH"""
     def visitor_fn(expression):
         if not isinstance(expression, BinaryComposition):
             return expression
         elif expression.operator == u'has_substring':
-            # Rewrite BinaryComposition expressions with "has_substring" into representable form.
```

Comment:
Let's keep one of these detailed comments as an example (I'd keep this one since it's the most detailed), and explain the slight differences compared to the lowering of `starts_with` and `ends_with`. Deleting all three of them seems like an unnecessary loss of documentation.

Reply:
sounds good, done